Scatter: default_opacities -> opacities

martinRenou opened the pull request:
Deprecate Scatter's `default_opacities` in favor of `opacities` (`default_colors` was deprecated in favor of `colors` in bqplot `0.8.4`).

Other marks had `opacities` instead of `default_opacities`, so this will make bqplot's API more homogeneous.

This is the first of several PRs for making `Mark`'s implementation more general, by adding common marks logic inside of it. For this purpose we need to make marks APIs more homogeneous.

martinRenou commented:
Small tip for the reviewer, use `?w=1` in the PR diff url in order to remove whitespaces changes: https://github.com/bloomberg/bqplot/pull/1011/files?w=1

martinRenou commented on js/src/Mark.ts:
```diff
@@ -208,6 +208,28 @@ export abstract class Mark extends widgets.WidgetView {
         return this.colors[index % len];
     }
 
+    get_mark_color(data, index) {
```
Those methods were in `ScatterBase.ts`, they will be very useful for other marks as well, not only for `Scatter`, so let's move them to the base class

martinRenou commented:
@SylvainCorlay If you have time reviewing :)

SylvainCorlay commented:
This is a longstanding inconsistency in bqplot.

I would be happy to see this go through although it is a backward incompatible change.

martinRenou commented:
This is not backward incompatible as far as I understand. I added a `default_opacities` property that prints a deprecated warning for now.

SylvainCorlay commented:
Oh great!

martinRenou commented:
Any change we can merge this PR?

martinRenou commented:
Something went wrong when I wanted to make the rebase. I'll reopen this.

maartenbreddels commented:
Since it is a breaking change, we should wait for 0.13 I'm afraid 😢

martinRenou commented:
Why do you say it's a breaking change?

I think we agreed with Sylvain it was not:
> This is not backward incompatible as far as I understand. I added a default_opacities property that prints a deprecated warning for now.

martinRenou commented:
Rebased again, and I updated the tests and the Notebook examples.

@SylvainCorlay do you think it should be merged or should we actually wait for 0.13?
